### DISCUSSION TITLE
How use swc_ecma_minifier

### DISCUSSION THREAD

**Discussion by JSerFeng** [swc-project/swc#4126] [Q&A] [answered]
Sorry I'm new to swc, how to use swc_ecma_minifier correctly?
My code represents something like this
```
optimize(
    ast,
    Arc::new(SourceMap::new(FilePathMapping::empty())),
    None,
    None,
    &MinifyOptions {
      rename: false,
      compress: config.compress.clone(),
      mangle: config.mangle.clone(),
      wrap: false,
      enclose: false,
    },
    &ExtraOptions {
      top_level_mark: Mark::from_u32(1111),
    },
)
```
and it panic: You should perform this operation in the closure passed to `set` of swc_common::syntax_pos::GLOBALS

**Comment by RiESAEX** [comment, marked as answer]
you need to wrap it with
```rust 
let globals = Globals::new();
GLOBALS.set(&globals,||{
// your code
})
```

**Reply by JSerFeng** [reply]
Thanks that solves my problem 😁

**Comment by jiesia** [comment]
I want to ask how to run minifier using multiple threads, because I found that it was actually running slowly.
